index.test.js

```diff
@@ -0,0 +1,156 @@
+const { deferred } = require('promise-callbacks');
+
+const Rewt = require('./');
+const redis = require('redis');
+
+describe('rewt', () => {
+  function buildRewt(url, namespace, ttl, cacheSecret) {
+    return new Rewt({
+      redisConn: redis.createClient(url),
+      redisNamespace: namespace,
+      ttl,
+      cacheSecret,
+    });
+  }
+
+  let rewt;
+  afterEach(() => {
+    if (rewt && rewt._redisConn) rewt._redisConn.quit();
+  });
+
+  describe('_generateKeyName', () => {
+    it('should generate the expected key name if provided', () => {
+      rewt = buildRewt('redis://localhost:6379', 'foobar');
+      expect(rewt._generateKeyName()).toEqual('foobar:rewt-secret');
+    });
+
+    it('should generate the expected key name if no namespace is provided', () => {
+      rewt = buildRewt('redis://localhost:6379');
+      expect(rewt._generateKeyName()).toEqual('rewt:rewt-secret');
+    });
+  });
+
+  if (process.env.INTEGRATION_TEST) {
```

Comment:
huh, where are these set? I think we can remove them given many of our libraries expect redis for testing and CI handles it :)

Reply:
These are very old - for this set, I agree, we should let people opt in, I do feel like we shouldn't require the timeout cache invalidation test (I'll setup a separate alias for running it).